lib/ar_state_machine.rb

```diff
@@ -121,6 +121,19 @@ def allow_transition?(states, from, to)
     states[from.to_sym].include?(to.to_sym)
   end
 
+  def set_state_by_id
```

Comment:
```ruby
def set_state_by_id(ss)
  if self.respond_to?("#{ss}_by_id")
    overwrite = true
    if self.respond_to?("overwrite_#{ss}_by_id")
      overwrite = !(self.send("overwrite_#{ss}_by_id") == false)
    elsif self.class.respond_to?("overwrite_#{ss}_by_id")
      overwrite = !(self.class.send("overwrite_#{ss}_by_id") == false)
    end
    if self.send("#{ss}_by_id").blank? or overwrite
      self.send("#{ss}_by_id=", self.last_edited_by_id)
    end
  end
end
```

then in the make_state functions... just need to do `set_state_by_id(ss)` after it sets self.last_edited_by_id